bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/ControllerHandler.java

```diff
@@ -44,116 +47,163 @@
+        cfg = getConfigAs(ControllerConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        determineTemperatureUnits();
+        stopPoller();
+        startPoller();
     }
 
     @Override
     public void dispose() {
-        stopRefresh();
-        logger.debug("Disconnecting CoolMasterNet Controller handler...");
+        updateStatus(ThingStatus.OFFLINE);
+        stopPoller();
         disconnect();
-        super.dispose();
     }
 
-    private void startRefresh() {
-        synchronized (refreshLock) {
-            ControllerConfiguration config = getConfigAs(ControllerConfiguration.class);
-            logger.debug("Scheduling new refresh job");
-            refreshJob = scheduler.scheduleWithFixedDelay(this::refreshHVACUnits, 0, config.refresh, TimeUnit.SECONDS);
+    /**
+     * Obtain the temperature unit configured for this controller.
+     *
+     * <p>
+     * CoolMasterNet defaults to Celsius, but allows a user to change the scale
+     * on a per-controller basis using the ASCII I/F "set deg" command. Given
+     * changing the unit is very rarely performed, there is no direct support
+     * for doing so within this binding.
+     *
+     * @return the unit as determined from the first line of the "ls" command
+     */
+    public Unit<?> getUnit() {
+        return unit;
+    }
+
+    private void determineTemperatureUnits() {
+        synchronized (socketLock) {
+            try {
+                checkConnection();
+                final String ls = sendCommand("ls");
+                if (ls.length() < LS_LINE_LENGTH) {
+                    throw new CoolMasterClientError("Invalid 'ls' response: '" + ls + "'");
+                }
+                final char scale = ls.charAt(LS_LINE_TEMP_SCALE_OFFSET);
+                unit = scale == 'C' ? SIUnits.CELSIUS : ImperialUnits.FAHRENHEIT;
+                logger.trace("Temperature scale '{}' set to {}", scale, unit);
+            } catch (final IOException ioe) {
+                logger.warn("Could not determine temperature scale", ioe);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ioe.getMessage());
+            }
+        }
+    }
+
+    private void startPoller() {
+        synchronized (scheduler) {
+            logger.debug("Scheduling new poller");
+            poller = scheduler.scheduleWithFixedDelay(this::poll, 0, cfg.refresh, SECONDS);
         }
     }
 
-    private void stopRefresh() {
-        synchronized (refreshLock) {
-            ScheduledFuture<?> localRefreshJob = refreshJob;
-            if (localRefreshJob != null && !localRefreshJob.isCancelled()) {
-                logger.debug("Cancelling existing refresh job");
-                localRefreshJob.cancel(true);
-                refreshJob = null;
+    private void stopPoller() {
+        synchronized (scheduler) {
+            final ScheduledFuture<?> poller = this.poller;
+            if (poller != null && !poller.isCancelled()) {
```

Comment:
There is no harm in cancelling an already cancelled task. Also I'm pretty sure you still want to set the poller back to null regardless.
```suggestion
            if (poller != null) {
```

Reply:
Will make this change.